.circleci/config.yml

```diff
@@ -0,0 +1,154 @@
+# Scala CircleCI 2.0 configuration file
+#
+# Check https://circleci.com/docs/2.0/sample-config/ for more details
+#
+# Use the chisel3-tools image to build and run FIRRTL and chisel tests.
+defaultImageUserWDEnv: &defaultImageUserWDEnv
+  docker:
+    - image: ucbbar/chisel3-tools
+      user: "chisel"
+
+  working_directory: ~/repo
+
+  environment:
+    # Customize the JVM maximum heap limit
+    JVM_OPTS: -Xmx3200m
+    TERM: dumb
+    CHISEL_REV: origin/master
+    FIRRTL_REPO: git@github.com:freechipsproject/firrtl.git
+    FIRRTL_BRANCH: master
+    FIRRTL_REV: master
+    SBT_ARGS: ""
+
+
+version: 2
+
+jobs:
+  build-prep:
+    <<: *defaultImageUserWDEnv
+
+    steps:
+
+    # Perform a "default" checkout to get ssh set up.
+      - checkout:
+          path: chisel3
+
+      - run:
+          command: |
+            date > date.prep
+            printenv
+            ls -l
+            git clone --depth 10 --branch $FIRRTL_BRANCH "$FIRRTL_REPO" firrtl && (cd firrtl && git checkout $FIRRTL_REV)
+            echo $FIRRTL_REV && (cd firrtl && git log -1 > ../firrtl.log)
+
+
+      - persist_to_workspace:
+          root: /home/chisel
+          paths:
+            - repo
+            - .ivy2
+            - .m2
+            - .sbt
+
+  build-firrtl:
+    <<: *defaultImageUserWDEnv
+
+    steps:
+      - attach_workspace:
+          at: /home/chisel
+
+      # publish FIRRTL
+      - run:
+          command: |
+            date > date.firrtl
+            (cd firrtl && cat /dev/null | sbt $SBT_ARGS +publishLocal)
```

Comment:
It's publishing 2.12 as well. From [build 23](https://circleci.com/gh/freechipsproject/chisel3/23)
```
 ...
[info] :: delivering :: edu.berkeley.cs#firrtl_2.12;1.2-SNAPSHOT :: 1.2-SNAPSHOT :: integration :: Thu Nov 01 17:12:09 UTC 2018
[info] 	delivering ivy file to /home/chisel/repo/firrtl/target/scala-2.12/ivy-1.2-SNAPSHOT.xml
[info] 	published firrtl_2.12 to /home/chisel/.ivy2/local/edu.berkeley.cs/firrtl_2.12/1.2-SNAPSHOT/poms/firrtl_2.12.pom
[info] 	published firrtl_2.12 to /home/chisel/.ivy2/local/edu.berkeley.cs/firrtl_2.12/1.2-SNAPSHOT/jars/firrtl_2.12.jar
[info] 	published firrtl_2.12 to /home/chisel/.ivy2/local/edu.berkeley.cs/firrtl_2.12/1.2-SNAPSHOT/srcs/firrtl_2.12-sources.jar
[info] 	published firrtl_2.12 to /home/chisel/.ivy2/local/edu.berkeley.cs/firrtl_2.12/1.2-SNAPSHOT/docs/firrtl_2.12-javadoc.jar
[info] 	published ivy to /home/chisel/.ivy2/local/edu.berkeley.cs/firrtl_2.12/1.2-SNAPSHOT/ivys/ivy.xml
[success] Total time: 78 s, completed Nov 1, 2018 5:12:10 PM
 ...
```
Otherwise, the chisel 2.12 tests wouldn't work.